### PR TITLE
acme_certificate: allow to download alternate certificate chains

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Background: Let's Encrypt [announced](https://letsencrypt.org/2019/04/15/transitioning-to-isrg-root.html) that they want to switch to the new ISRG root certificate this summer, i.e. they will start delivering the intermediate certificate signed by their own root and not by the IdenTrust root. The consequence is that Let's Encrypt certs using this chain won't be supported by a lot of older devices and browsers which support the IdenTrust root, but not the ISRG root. (Support for the IdenTrust root is listed [here](https://letsencrypt.org/docs/certificate-compatibility/), there is unfortunately no similar listing of devices/browsers supporting the new root, except the statement [here](https://letsencrypt.org/2018/08/06/trusted-by-all-major-root-programs.html).) There have been some discussions about this ([here](https://community.letsencrypt.org/t/please-reconsider-defaulting-to-the-isrg-root-its-unsupported-by-more-than-50-of-android-phones/91485), [here](https://community.letsencrypt.org/t/please-update-the-certificate-compatibility-page-to-include-information-about-the-new-isrg-roots/91436/19)). **Update:** the root change has been [delayed by one year](https://community.letsencrypt.org/t/transitioning-to-isrgs-root/94056/1).

Anyway, the ACME protocol offers [a way for CAs to offer alternative chains](https://tools.ietf.org/html/rfc8555#section-7.4.2), and my hope is that Let's Encrypt will use that feature of the protocol to still deliver the old intermediate certificate. That would allow ACME clients to offer the "old" chain to the users without ugly hacks (see [here](https://community.letsencrypt.org/t/please-update-the-certificate-compatibility-page-to-include-information-about-the-new-isrg-roots/91436/13) for an example).

This is an experimental branch to play around with downloading alternate chains. Works so far with letsencrypt/pebble#234

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
acme_certificate

### CODE REVIEW

**Comment by felixfontein** [comment]
Rebased to include #58294.

**Comment by felixfontein** [comment]
Ok, I'm declaring this as done. :)

I later want to add another option, which allows to select the chain to write to disk using a priority list of key IDs / fingerprints of intermediate / root certificates. But for now, already having the possibility to extract all chains is a good start.

**Comment by felixfontein** [comment]
ready_for_review

**Review by resmo — APPROVED** [review]
LGTM

File: lib/ansible/modules/crypto/acme/acme_certificate.py [hunk 1]
```diff
@@ -197,6 +197,15 @@
     type: bool
     default: no
     version_added: 2.6
+  retrieve_all_alternates:
+    description:
+      - "When set to C(yes), will retrieve all alternate chains offered by the ACME CA.
+         These will not be written to disk, be returned together with the main chain as
```
Comment (ctrufan) [inline lib/ansible/modules/crypto/acme/acme_certificate.py]: "not be written to disk, be returned"
"not be written to disk, but be returned" or "not be written to disk. They will be returned"

Comment (felixfontein) [reply lib/ansible/modules/crypto/acme/acme_certificate.py]: Good point!

Comment (felixfontein) [reply lib/ansible/modules/crypto/acme/acme_certificate.py]: Changed it to "[...] to disk, but will be returned".

**Comment by felixfontein** [comment]
@resmo @ctrufan thanks for reviewing this! :)

**Pull request merged**